tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -590,50 +589,67 @@ static unique_ptr<TableFunctionRef> ScanReplacement(ClientContext &context, cons
 	return nullptr;
 }
 
-shared_ptr<DuckDBPyConnection> DuckDBPyConnection::Connect(const string &database, bool read_only,
-                                                           const py::dict &config_dict) {
-	auto res = make_shared<DuckDBPyConnection>();
-
-	DBConfig config;
-
-	if (read_only) {
-		config.options.access_mode = AccessMode::READ_ONLY;
+string GetDBAbsolutePath(const string &database) {
+	if (database.empty()) {
+		return ":memory:";
 	}
-	for (auto &kv : config_dict) {
-		string key = py::str(kv.first);
-		string val = py::str(kv.second);
-		auto config_property = DBConfig::GetOptionByName(key);
-		if (!config_property) {
-			throw InvalidInputException("Unrecognized configuration property \"%s\"", key);
-		}
-		config.SetOption(*config_property, Value(val));
-	}
-	res->database = make_unique<DuckDB>(database, &config);
-	res->connection = make_unique<Connection>(*res->database);
-	if (config.options.enable_external_access) {
-		res->database->instance->config.replacement_scans.emplace_back(ScanReplacement);
+	if (database.rfind(":memory:", 0) == 0) {
+		// this is a memory db, just return it.
+		return database;
 	}
+	py::object abspath = py::module_::import("os.path").attr("abspath");
+	return py::str(abspath(database));
```

Comment:
Do you think it would be difficult to do this inside the DB instance cache itself (without calling a Python method)? It would be nicer if this could be done for all clients in the same manner.